fenix/app/src/androidTest/java/org/mozilla/fenix/ui/FirefoxSuggestTest.kt

```diff
@@ -33,22 +35,86 @@ class FirefoxSuggestTest {
         ),
     ) { it.activity }
 
+    private val sponsoredKeyWords: Map<String, List<String>> =
+        mapOf(
+            "Amazon" to
+                listOf(
+                    "Amazon.com - Official Site",
+                    "amazon.com/?tag=admarketus-20&ref=pd_sl_924ab4435c5a5c23aa2804307ee0669ab36f88caee841ce51d1f2ecb&mfadid=adm",
+                ),
+            "Nike" to
+                listOf(
+                    "Nike.com - Official Site",
+                    "nike.com/?cp=16423867261_search_318370984us128${getSponsoredFxSuggestPlaceHolder()}&mfadid=adm",
+                ),
+            "Macy" to listOf(
+                "macys.com - Official Site",
+                "macys.com/?cm_mmc=Google_AdMarketPlace-_-Privacy_Instant%20Suggest-_-319101130_Broad-_-kclickid__kenshoo_clickid_&m_sc=sem&m_sb=Admarketplace&m_tp=Search&m_ac=Admarketplace&m_ag=Instant%20Suggest&m_cn=Privacy&m_pi=kclickid__kenshoo_clickid__319101130us1201${getSponsoredFxSuggestPlaceHolder()}&mfadid=adm",
+            ),
+            "Spanx" to listOf(
+                "SPANX® -  Official Site",
+                "spanx.com/?utm_source=admarketplace&utm_medium=cpc&utm_campaign=privacy&utm_content=319093361us1202${getSponsoredFxSuggestPlaceHolder()}&mfadid=adm",
+            ),
+            "Bloom" to listOf(
+                "Bloomingdales.com - Official Site",
+                "bloomingdales.com/?cm_mmc=Admarketplace-_-Privacy-_-Privacy-_-privacy%20instant%20suggest-_-319093353us1228${getSponsoredFxSuggestPlaceHolder()}-_-kclickid__kenshoo_clickid_&mfadid=adm",
+            ),
+            "Groupon" to listOf(
+                "groupon.com - Discover & Save!",
+                "groupon.com/?utm_source=google&utm_medium=cpc&utm_campaign=us_dt_sea_ggl_txt_smp_sr_cbp_ch1_nbr_k*{keyword}_m*{match-type}_d*ADMRKT_319093357us1279${getSponsoredFxSuggestPlaceHolder()}&mfadid=adm",
+            ),
+        )
```

Comment:
Verify hard for me to review this because I don't know enough about these. These links seem highly structured with campaign details and such. Is there any way to pull these rather than hard-code? One single character change in the URLs and these break.

Reply:
Forgot to add in PR the details the [list](https://drive.google.com/file/d/1o2nLj1L1IAHZCSUL08QHd7sJCEpS8ZZ0/view) based on which, each keyword points to a sponsored website and the corresponding link.

Will check to see if these remain the same or could be changed at any point.

Reply:
I love the approach of using these to cycle through your tests. It's the first step in a data-driven approach to dynamically create your tests through self-discoverability.

The next steps would be generating this data from a source of truth, like the text doc where all these sponsored links live.

Something to consider for doing that would be either starting with saving that doc as a fixture, parsing it, and then pulling in, say, the first 5 links from each category, or something like that; lots of different approaches work there.

If that source of truth is hosted somewhere, you could make a call to it and do that same parsing. That's typically a third step as it adds a hosting/network dependency you need to defend against.

All-in-all, I think it's a great entry-point into some more dynamic, eventually self-writing test automation that will get more coverage for less code and much less maintenance costs. 👍